File: 1768. Merge Strings Alternately/solution.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int countStudents(std::vector<int>& students, std::vector<int>& sandwiches) {
        int lenCal = 0;
        std::vector<int> stu(students.begin(), students.end());
        std::vector<int> san(sandwiches.begin(), sandwiches.end());
        
        while (!stu.empty() && lenCal <= stu.size()) {
            if (stu[0] == san[0]) {
                stu.erase(stu.begin());
                san.erase(san.begin());
                lenCal = 0;
            } else {
                stu.push_back(stu[0]);
                stu.erase(stu.begin());
                lenCal++;
            }
        }
        return stu.size();
    }
};

int main() {
    std::vector<int> stu = {1, 1, 1, 0, 0, 1};
    std::vector<int> san = {1, 0, 0, 0, 1, 1};
    Solution ss;
    int result = ss.countStudents(stu, san);
    std::cout << result << std::endl;
    return 0;
}

File: 1768. Merge Strings Alternately/solution.java
import java.util.ArrayList;

class solution {
    public int countStudents(int[] students, int[] sandwiches) {
        int lenCal= 0;
        ArrayList<Integer> stu = new ArrayList<>();
        ArrayList<Integer> san = new ArrayList<>();
        for(int s : students)
        {
            stu.add(s);
        }
        for(int s : sandwiches)
        {
            san.add(s);
        }
        while(stu.size() > 0 && lenCal <=  stu.size())
        {
            if(stu.get(0) ==  san.get(0))
            {
                stu.remove(0);
                san.remove(0);
                lenCal = 0;
            }
            else
            {
                stu.add(stu.get(0));
                stu.remove(0);
                lenCal++;
            }
        }
        return stu.size();
    }
    public static void main(String[] args)
    {
        int[] stu = {1,1,1,0,0,1};
        int[] san = {1,0,0,0,1,1};
        solution ss = new solution();
        int result = ss.countStudents(stu, san);
        System.out.println(result);
    }
}